zingolib/src/lightclient/describe.rs

```diff
@@ -481,19 +511,23 @@ impl LightClient {
                     });
                 }
             }
-            // We spent funds, and received funds as non-change. This is most likely a send-to-self,
-            // TODO: Figure out what kind of special-case handling we want for these
-            (true, true) => {
+            // If the transaction is outgoing, and there are notes in:
+            //   - sapling notes
+            //   - orchard notes
+            // then they were sent to self
+            if !transaction_record.sapling_notes.is_empty()
+                || !transaction_record.orchard_notes.is_empty()
+            {
```

Comment:
this will create a `SendToSelf` value transfer for every change note received which is not the correct behaviour we currently have or want. Currently, we have this idea of send-to-selfs not being marked as change and we only display non-change where we spent funds to create the tx as "send-to-self". I think we want to define change as ANY output we receive in a tx created by our capability, so therefore we need to define a send-to-self as an output from a transaction we created with no outgoing_tx_data

Reply:
Really good point @Oscar-Pepper!  I am glad you thought through back-compatibility more carefully than I did.
I feel like this is something to reconsider in the new architecture.

Reply:
Resolved, pending a local test pass.